mlflow/pyfunc/model.py

```diff
@@ -391,19 +391,330 @@ def predict_stream(
+
+    - Adds the ``attachments`` key in every input/output message for tools and internal agent calls so they can return additional outputs such as visualizations and progress indicators
+    - Adds the ``context`` key with a ``conversation_id`` and ``user_id`` to enable modifying the behavior of the agent depending on the user querying the agent
+    - Adds the ``custom_inputs`` key, an arbitrary ``dict[str, Any]`` to pass in any additional information to modify the agent's behavior
+
+    The :py:class:`ChatAgentResponse <mlflow.types.agent.ChatAgentResponse>` schema diverges from :py:class:`ChatCompletionResponse <mlflow.types.llm.ChatCompletionResponse>` schema in the following ways:
+
+    - Adds the ``custom_outputs`` key, an arbitrary ``dict[str, Any]`` to return any additional information
+    - Allows multiple messages in the output, to improve the  display and evaluation of internal tool calls and inter-agent communication that led to the final answer.
+
+    Here's an example of a :py:class:`ChatAgentResponse <mlflow.types.agent.ChatAgentResponse>` detailing a tool call:
+
+    .. code-block:: python
+
+        {
+            "messages": [
+                {
+                    "role": "assistant",
+                    "content": "",
+                    "id": "run-04b46401-c569-4a4a-933e-62e38d8f9647-0",
+                    "tool_calls": [
+                        {
+                            "id": "call_15ca4fcc-ffa1-419a-8748-3bea34b9c043",
+                            "type": "function",
+                            "function": {
+                                "name": "generate_random_ints",
+                                "arguments": '{"min": 1, "max": 100, "size": 5}',
+                            },
+                        }
+                    ],
+                },
+                {
+                    "role": "tool",
+                    "content": '{"content": "Successfully generated array of 5 random ints in [1,
+                    100].", "custom_outputs": {"random_nums": [93, 51, 12, 7, 25]}}',
+                    "name": "generate_random_ints",
+                    "id": "call_15ca4fcc-ffa1-419a-8748-3bea34b9c043",
+                    "tool_call_id": "call_15ca4fcc-ffa1-419a-8748-3bea34b9c043",
+                },
+                {
+                    "role": "assistant",
+                    "content": "The new set of generated random numbers are: 93, 51, 12, 7, and 25. Would you like me to generate more?",
+                    "name": "llm",
+                    "id": "run-70c7c738-739f-4ecd-ad18-0ae232df24e8-0",
+                },
+            ],
+            "custom_outputs": {"random_nums": [93, 51, 12, 7, 25]},
+        }
+
+    **Streaming Agent Output with ChatAgent**
+
+    The ChatAgent interface has a few streaming specific behaviors that are required by us and assumed by our helper LangChainChatAgent and LangGraphChatAgent classes.
```

Comment:
```suggestion
    To support streaming the output of your agent, implement `predict_stream` in your subclass of `ChatAgent`. When implementing `predict_stream`, keep in mind the following requirements:
```

Also - should/could we move these behavior requirements to the `predict_stream` docstring and link them from here?